packages/nimble-components/src/table/tests/table-keyboard-navigation.spec.ts

```diff
@@ -0,0 +1,917 @@
+    return fixture<Table<SimpleTableRecord>>(
+        html`<nimble-table id-field-name="id">
+        <nimble-table-column-text id="first-column" field-name="stringData1" column-id="column-1"></nimble-table-column-text>
+        <nimble-table-column-text id="second-column" field-name="stringData2" column-id="column-2"></nimble-table-column-text>
+        <nimble-table-column-text id="third-column" field-name="stringData3" column-id="column-3"></nimble-table-column-text>
+    </nimble-table>`
+    );
+}
+
+// prettier-ignore
+async function setupWithAnchorColumns(): Promise<Fixture<Table<SimpleTableRecord>>> {
+    return fixture<Table<SimpleTableRecord>>(
+        html`<nimble-table id-field-name="id">
+        <nimble-table-column-anchor id="first-column" href-field-name="href1" label-field-name="href1" column-id="column-1"></nimble-table-column-anchor>
+        <nimble-table-column-anchor id="second-column" href-field-name="href2" label-field-name="href2" column-id="column-2"></nimble-table-column-anchor>
+        <nimble-table-column-anchor id="third-column" href-field-name="href3" label-field-name="href3" column-id="column-3"></nimble-table-column-anchor>
+    </nimble-table>`
+    );
+}
+
+function currentFocusedElement(): HTMLElement | null {
+    let activeElement = document.activeElement;
+    while (activeElement?.shadowRoot?.activeElement) {
+        activeElement = activeElement.shadowRoot.activeElement;
+        // In some cases, the active element may be a sub-part of a control (example: MenuButton -> ToggleButton -> a div with tabindex=0). Stop at the outer control boundary, so that
+        // we can more simply check element equality.
+        // (For rows/cells/cell views, we do need to recurse into them, to get to the appropriate focused controls though)
+        if (
+            activeElement instanceof FoundationElement
+            && !(activeElement instanceof TableRow)
+            && !(activeElement instanceof TableCell)
+            && !(activeElement instanceof TableCellView)
+        ) {
+            break;
+        }
+    }
+
+    return activeElement as HTMLElement;
+}
+
+describe('Table keyboard navigation', () => {
+    let element: Table<SimpleTableRecord>;
+    let connect: () => Promise<void>;
+    let disconnect: () => Promise<void>;
+    let pageObject: TablePageObject<SimpleTableRecord>;
+
+    async function addActionMenu(column: TableColumn): Promise<void> {
+        column.actionMenuSlot = `${column.id}-action-menu-slot`;
+        column.actionMenuLabel = 'Actions';
+
+        const menu = document.createElement('nimble-menu');
+        menu.slot = column.actionMenuSlot;
+
+        const menuItem1 = document.createElement('nimble-menu-item');
+        menuItem1.textContent = 'menu item 1';
+        menu.appendChild(menuItem1);
+
+        element.appendChild(menu);
+
+        await waitForUpdatesAsync();
+    }
+
+    function isCellActionMenuVisible(
```

Comment:
Any reason to not put this in the pageObject?

Reply:
Actually there was already an equivalent page object method that I missed. Switched to it and removed this.